manifests/config.pp

```diff
@@ -12,14 +12,132 @@
 #   ident any configuration added in this iteration of the
 #   recursion. Defauls to 0 
 # @param values
-#   Hash of configuration parameters to include in $filename
+#   The hash it expects needs name which corresponds to the $filename saved in $config_path/$local_configdir/
+#   For instance the following hash will produce a file in /etc/dovecot/conf.d/master.conf if $config_path
+#   and $local_configdir are set to default.
+#   **NOTE - the file named will be completely overwritten, so ensure that ALL needed values are specified.**
+#
+#   <a name="values_example"></a>`Example (hiera):`
+#
+#   ```
+#   dovecot::config:
+#     'master.conf':                     # <- name ($filename)
+#       values:
+#         default_process_limit: 350
+#         default_vsz_limit: 1024M
+#         default_client_limit: 2000
+#   ```
+#
+#   The resulting **/etc/dovecot/conf.d/master.conf** will look like this:
+#
+#   ```
+#   This file is managed by Puppet. DO NOT EDIT.
+#   default_client_limit = 2000
+#   default_process_limit = 350
+#   default_vsz_limit = 1024M
+#   ```
+#   
 #   Defaults to {}
-#   see ::dovecot for more information
 # @param sections
-#   Array of configuration section to include in $filenmame
-#   Defaults to []
-#   see ::dovecot for more information
+#   Sometimes you need to have [Sections](https://doc.dovecot.org/configuration_manual/config_file/#sections) 
+#   in your config files.  
+#   These are defined as an Array of hashes similar to the intial config hash:  
+#   Expanding on our previous example, wanting to add an section we can add a 'sections' key to our hash.  
+#   Each additional section is started with a **-** on its own line and indented correctly.
+#
+#   <a name="sections_example"></a>`Example (hiera):`
+#
+#   ```
+#   dovecot::config:
+#     'mail.conf':
+#       values:
+#         'mail_location': 'maildir:~/'
+#       sections:
+#         -                        # <- Section starts ( $name { $values } )
+#           name: 'namespace inbox'
+#           values:
+#             'inbox': 'yes'
+#             'seperator': '.'
+#             'prefix': 'INBOX'
+#   ``` 
+#
+#   This will result in **/etc/dovecot/conf.d/mail.conf** containing the following:
 #
+#   ```
+#   This file is managed by Puppet. DO NOT EDIT.
+#   mail_location = maildir:~/
+#   namespace inbox {
+#     inbox = yes
+#     separator =.
+#     prefix =INBOX.
+#   }
+#   ``` 
+#
+#   Some dovecot sections have a double bracket system (section within a section). This is done as follows:
+#
+#   Example (hiera):
+#
+#   ```
+#   dovecot::config:
+#     'master.conf':
+#       values:
+#         default_process_limit: 350
+#         default_vsz_limit: 1024M
+#         default_client_limit: 2100
+#       sections:
+#         -
+#           name: 'service imap-login'
```

Comment:
Line 88 and 89 can be combined to
```- name: 'service imap-login'```

which should also avoid some newlines in the resulting file.